src/components/shared/data-annotation/DataAnnotation.js

```diff
@@ -2,10 +2,36 @@
 /* eslint-disable react/jsx-filename-extension */
 import React from 'react';
 import fontSizes from '../../../constants/fontSizes';
+import colors from '../../../constants/colors';
+
+function DataAnnotation({
+  x,
+  y,
+  width,
+  height,
+  offset,
+  value,
+  valueFormatter,
+  position,
+  shouldUseSpaceOffset,
+}) {
+  const positionAttributes = {
+    top: {
+      x: shouldUseSpaceOffset ? x + width / 2 : x,
+      y: y - offset,
+      textAnchor: 'middle',
+      verticalAnchor: 'end',
+    },
+    right: {
+      x: x + width + offset,
+      y: shouldUseSpaceOffset ? y + height / 2 : y,
+      textAnchor: 'start',
+      verticalAnchor: 'middle',
+    },
+  }[position];
 
-function DataAnnotation({ x, y, stroke, value, valueFormatter }) {
   return (
-    <text x={x} y={y} dy={-4} fill={stroke} fontSize={fontSizes['2xs']} textAnchor="middle">
+    <text x={x} y={y} {...positionAttributes} fill={colors.gray[800]} fontSize={fontSizes['2xs']}>
```

Comment:
Got rid of the `dy={-4}` that line plots needed by doing `verticalAnchor: 'end'` now, same as how `recharts` is doing it from the links in the comment above.